sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -50,6 +50,7 @@ trait InvokeLike extends Expression with NonSQLExpression with ImplicitCastInput
 
   def propagateNull: Boolean
 
+  override def foldable: Boolean = children.forall(_.foldable) && deterministic
```

Comment:
shall we remove `deterministic`? I don't see other expressions require it when defining foldable.

Reply:
> shall we remove `deterministic`? I don't see other expressions require it when defining foldable.

If it's not `deterministic`, such as Invoke return a rand result, if we fold this, we may return incorrect answer